app/models/event.rb

```diff
@@ -209,6 +227,21 @@ def calculate_progress
 
   private
 
+  ##
+  # If max_attendees variable is set (higher than 0)
+  # variable require_registration must also be set
+  def max_attendees_and_require_registration
+    errors.add(:require_registration, 'must be enabled, when you set max_attendees') if max_attendees && !require_registration
+    errors.add(:max_attendees, 'must be enabled, when you set require_registration') if require_registration && max_attendees.nil?
+  end
+
+  ##
+  # Do not allow, for the event, more attendees than the size of the room
+  def max_attendees_no_more_than_room_size
```

Comment:
We can overcome this by only adding a validation error if the attribute max_attendees has changed (which does not happen when scheduling).